src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/TestServiceContainerRealRefPass.php

```diff
@@ -29,15 +30,35 @@ public function process(ContainerBuilder $container)
         $privateContainer = $container->getDefinition('test.private_services_locator');
         $definitions = $container->getDefinitions();
         $privateServices = $privateContainer->getArgument(0);
+        $renamedIds = [];
 
         foreach ($privateServices as $id => $argument) {
             if (isset($definitions[$target = (string) $argument->getValues()[0]])) {
                 $argument->setValues([new Reference($target)]);
+                if ($id !== $target) {
+                    $renamedIds[$id] = $target;
+                }
             } else {
                 unset($privateServices[$id]);
             }
         }
 
+        foreach ($container->getAliases() as $id => $target) {
+            while ($container->hasAlias($target = (string) $target)) {
```

Comment:
`do {} while()` to call `hasAlias` 1 less time?

Reply:
wouldn't work, read again ;)